shared/health/src/main/java/io/pravega/shared/health/HealthService.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.shared.health;
+
+/**
+ * The top level interface used to provide any and all health related information for a particular component
+ * of Pravega. It provides the {@link HealthEndpoint} used to make health
+ * information accessible to clients.
+ */
+public interface HealthService extends AutoCloseable {
+
+    /**
+     * The main interface between some client and the {@link HealthService}. The {@link HealthEndpoint} encapsulates
+     * the various types of requests the {@link HealthService} will be able to fulfill.
+     *
+     * @return The {@link HealthEndpoint} instance.
+     */
+    HealthEndpoint getEndpoint();
+
+    /**
+     * Provides the {@link HealthServiceUpdater} that currently performs the background {@link Health} requests (of the root).
+     *
+     * @return The  {@link HealthServiceUpdater}.
+     */
+    HealthServiceUpdater getHealthServiceUpdater();
```

Comment:
Is it necessary to expose the `HealthServiceUpdator` to Clients? What would clients do with it? I see only 2 public APIs on this class: `getLatestHealth()` and `getDuration()`. 
**Do we want to allow clients to trigger a health probe synchronously?** If not, we should not be exposing this class.
Classes exposed to clients become part of the public API and are difficult to modify later. I would not recommend doing this unless absolutely necessary.

Reply:
The current reason why the `HealthServiceUpdater` is exposed is to allow clients to properly start and stop the underlying executor services. I'm not quite sure how we would handle those cases if made private.